cmd/mdatagen/templates/component_test.go.tmpl

```diff
@@ -224,3 +230,72 @@ func Test_ComponentLifecycle(t *testing.T) {
 	}
 }
 {{ end }}
+
+{{ if isReceiver }}
+func Test_ComponentLifecycle(t *testing.T) {
```

Comment:
Doesn't seem like it's generated for the splunkhecreceiver?

Reply:
Don't we have a CI check to validate whether it's properly generated? If so we need to add another `git diff` check in `test-and-build.yml::checks` job.

Reply:
It doesn't because the file is not even checked in (adding it now) and therefore ignored by git diff.